src/main/java/com/epam/ta/reportportal/core/launch/impl/FinishLaunchHandlerImpl.java

```diff
@@ -90,25 +92,26 @@ public FinishLaunchHandlerImpl(LaunchRepository launchRepository, TestItemReposi
 	}
 
 	@Override
-	public Launch finishLaunch(Long launchId, FinishExecutionRQ finishLaunchRQ, ReportPortalUser.ProjectDetails projectDetails,
+	public Launch finishLaunch(String launchId, FinishExecutionRQ finishLaunchRQ, ReportPortalUser.ProjectDetails projectDetails,
 			ReportPortalUser user) {
-		Launch launch = launchRepository.findById(launchId)
+		Launch launch = launchRepository.findByUuid(launchId)
 				.orElseThrow(() -> new ReportPortalException(LAUNCH_NOT_FOUND, launchId.toString()));
 
 		validateRoles(launch, user, projectDetails);
 		validate(launch, finishLaunchRQ);
 
 		Optional<StatusEnum> status = StatusEnum.fromValue(finishLaunchRQ.getStatus());
 
+		Long id = launch.getId();
 		if (launchRepository.hasItemsInStatuses(launch.getId(), Lists.newArrayList(JStatusEnum.IN_PROGRESS))) {
```

Comment:
Is it necessary to extract launch id as a variable?

Reply:
the reason was lambda expression in else block below - use launch.getId() there will require launch to be effectively final. so just extract it for simplicity